### ISSUE TITLE
some questions of "select outliers for each MDS PC" code

### ISSUE THREAD

**Issue by Axolotl233** [owensgl/haploblocks#1]
Hi
Recently, I used local_pca analysis and want to find some haplotype block like sunflower publication, so I refer to your code.But here are some things I don’t understand in **lostruct.annuus.ha412.R** ```202```-```215```
```
  if (nrow(pos_windows) >= min_windows){
    permutations <- matrix( nrow = n_permutations, ncol = 1)
    for (i in 1:n_permutations){
      max_1 <- win.regions %>% sample_n(nrow(pos_windows)) %>% group_by(chrom) %>% summarize(count=n()) %>% arrange(desc(count)) %>% head(n=1) %>% 
        ungroup() %>% summarize(sum=sum(count)) %>% pull(sum)
      permutations[i,1] <- max_1
    }
    pos_windows %>% group_by(chrom) %>% summarize(count=n()) %>% arrange(desc(count)) %>% head(n=1) %>% 
     ungroup() %>% summarize(sum=sum(count)) %>% pull(sum) -> sampled_max_1
    pos_windows %>% group_by(chrom) %>% summarize(count=n()) %>% arrange(desc(count)) %>% head(n=1) %>%
      pull(chrom) -> clustered_chr
    
    x <- as.tibble(permutations) %>%  filter(V1 >= sampled_max_1) %>% nrow() 
    x <- x+1
   pvalue <- x/(n_permutations+1)
```
please tell the principes p-value calculate methods here. If i have just one chromosome result, the V1 will be equal to sampled_max_1, how can i calculate p-value in this condition?

i 'm a newble of R and population genetics, If you could kindly provide me some advice or code, I will be appreciate.

**Comment by owensgl** [comment]
Hi, 
The code you are highlighting was part of a work in progress where I tried a few different ways to pull out these outlier regions. Nothing I tried actually work very well at pulling out the outlier regions automatically, because sometimes they were in multiple discrete chunks and also there were other signals that were similarly high but more gradual across the genome. So for the paper we plotted the top 20 MDS axes values across the chromosome for each chromosome separately and then manually picked out the outliers. From that I picked a threshold MDS value to classify a point as an outlier or not and used some code to fill in gaps where there was a non-outlier window sandwiched between outlier windows. 

I'm sure someone could find a way to find these automatically, but I didn't in this paper, so my recommendation is just plotting and manually annotating.
